### PR TITLE
feat(logs): add --timestamp, --follow and --colorize flags

### PR DESCRIPTION
**Pull request by erfanium**
Hello all.
I have added --timestamp and --follow flags to `liara logs` command.  
these two flags will exactly behave like docker's --timestamp` and --follow flags and will make liara cli more like docker cli.

Also sometimes fetching logs from server takes more than 1 second, so i added `pendingFetch` to prevent the cli from sending simultaneous requests to the server which could result in the display of duplicated logs.

### CODE REVIEW

**Comment by mhemrg** [comment]
We should also enable `follow` and `timestamps` by default here:

https://github.com/liara-cloud/liara-cli/blob/75a9f4e0d9b51475568067498fc631125ea93fd4/src/commands/deploy.ts#L187-L197

**Comment by erfanium** [comment]
@mhemrg done

**Comment by mhemrg** [comment]
Great job, Erfan.

Changing the default behavior of this command is a breaking change. But we're also aiming for Docker CLI compatibility.
What do you think @guoard?

**Comment by erfanium** [comment]
Let me explain some real-world use cases of this PR tp show why current liara logs command is not well designed.

1. some apps/frameworks are using json logs. (e.g. fastify, pino). I need to pipe the logs into another json processing tool. (like jq). so I would need an option to see my actual app logs without extra modification: no timestamp, no colorizing. with respect to stdin/stderr.

```sh
liara logs -a=myapp -f | jq
```

2. currently there is no option to disable the follow behavior of liara logs. so it's not easy to save the logs into a file. currently this command will never exit:
```sh
liara logs -a=myapp > app.log
````

**Comment by erfanium** [comment]
I have another personal suggestion you can think of it.
`colorfulAccessLog` is extra here. this can be a considered as a cute feature but it will silently changed the app log and can introduce problems. also it's not part of docker

**Comment by mhemrg** [comment]
What if we add another flag to enable colorized output?

**Review by guoard — APPROVED** [review]
@erfanium's explanations of the use cases are both reasonable and well-thought-out and I don't foresee any issues with releasing a new major version.

**Comment by erfanium** [comment]
new changes:
- I have added --colorize flag
- timestamp are now formatted in isoDateString. to be align be with docker

File: README.md [hunk 1]
```diff
@@ -309,12 +309,16 @@ fetch the logs of an app
...
+  $ liara app logs [-h] [--debug] [--api-token <value>] [--region iran|germany] [-a <value>] [-s <value>] [-t]
+    [-f] [-c]
 
 FLAGS
   -a, --app=<value>    app id
+  -c, --colorize       Colorize log output
+  -f, --follow         Follow log output
   -h, --help           Show CLI help.
   -s, --since=<value>  show logs since timestamp
+  -t, --timestamp      Show timestamps
```
Comment (mhemrg) [inline README.md:321]: Let's use the plural version: https://docs.docker.com/engine/reference/commandline/logs/#options

```suggestion
  -t, --timestamps      Show timestamps
```

File: src/commands/app/logs.ts [hunk 2]
```diff
@@ -22,21 +22,47 @@ export default class AppLogs extends Command {
...
 
   static aliases = ['logs'];
 
+  #timestamp = false;
+  #colorize = false;
+
   async run() {
     const { flags } = await this.parse(AppLogs);
     let since: string | number = flags.since || 1;
+    const { follow, colorize, timestamp } = flags;
```
Comment (mhemrg) [inline src/commands/app/logs.ts:50]:
```suggestion
    const { follow, colorize, timestamps } = flags;
```

File: src/commands/app/logs.ts [hunk 3]
```diff
@@ -22,21 +22,47 @@ export default class AppLogs extends Command {
       char: 's',
       description: 'show logs since timestamp',
     }),
+    timestamp: Flags.boolean({
```
Comment (mhemrg) [inline src/commands/app/logs.ts:25]:
```suggestion
    timestamps: Flags.boolean({
```

File: src/commands/app/logs.ts [hunk 4]
```diff
@@ -72,12 +98,38 @@ Sorry for inconvenience. Please contact us.`).render()
...
+    return chalk.gray(message);
+  }
+
+  #printLogLine(log: ILog) {
+    let message = log.message;
+    if (this.#colorize) {
+      message = colorfulAccessLog(message);
+    }
+
+    if (this.#timestamp) {
```
Comment (mhemrg) [inline src/commands/app/logs.ts:126]:
```suggestion
    if (this.#timestamps) {
```

File: src/commands/deploy.ts [hunk 5]
```diff
@@ -190,6 +190,9 @@ export default class Deploy extends Command {
         config.app,
         '--since',
         moment().unix().toString(),
+        '--follow',
+        '--timestamp',
```
Comment (mhemrg) [inline src/commands/deploy.ts:194]:
```suggestion
        '--timestamps',
```

File: src/commands/app/logs.ts [hunk 6]
```diff
@@ -22,21 +22,47 @@ export default class AppLogs extends Command {
...
 
+  #timestamp = false;
+  #colorize = false;
+
   async run() {
     const { flags } = await this.parse(AppLogs);
     let since: string | number = flags.since || 1;
+    const { follow, colorize, timestamp } = flags;
+
+    this.#timestamp = timestamp;
```
Comment (mhemrg) [inline src/commands/app/logs.ts:52]:
```suggestion
    this.#timestamps = timestamps;
```

**Review by mhemrg — CHANGES_REQUESTED** [review]

**Comment by mhemrg** [comment]
Everything looks good to me @erfanium.
Let's just make the `timestamps` flag compatible with Docker CLI. We can then proceed to merge this.

**Comment by mhemrg** [comment]
And just a quick note worth mentioning: don't bother updating the readme. Before publishing the new version to npm, it will be generated automatically.

**Comment by erfanium** [comment]
@mhemrg done

File: src/commands/app/logs.ts [hunk 7]
```diff
@@ -22,21 +22,47 @@ export default class AppLogs extends Command {
       char: 's',
       description: 'show logs since timestamp',
     }),
+    timestamps: Flags.boolean({
+      char: 't',
+      description: 'Show timestamps',
```
Comment (mhemrg) [inline src/commands/app/logs.ts:27]: To be consistent with the rest of the project, let's use lowercase descriptions:
```suggestion
      description: 'show timestamps',
```

File: src/commands/app/logs.ts [hunk 8]
```diff
@@ -22,21 +22,47 @@ export default class AppLogs extends Command {
...
+      default: false,
+    }),
+    follow: Flags.boolean({
+      char: 'f',
+      description: 'Follow log output',
```
Comment (mhemrg) [inline src/commands/app/logs.ts:32]:
```suggestion
      description: 'follow log output',
```

File: src/commands/app/logs.ts [hunk 9]
```diff
@@ -22,21 +22,47 @@ export default class AppLogs extends Command {
...
+      default: false,
+    }),
+    colorize: Flags.boolean({
+      char: 'c',
+      description: 'Colorize log output',
```
Comment (mhemrg) [inline src/commands/app/logs.ts:37]:
```suggestion
      description: 'colorize log output',
```

**Review by mhemrg — CHANGES_REQUESTED** [review]
Hi @erfanium,
I've been using this PR lately and it works fine. Just asked for a few tweaks and then we're good to go ahead and merge this.

**Pull request merged**